### PR TITLE
feat: add Scaleway provider

### PR DESCRIPTION
**Pull request by dirien**
Scaleway provider implementation

PR for #83 

Caddy implementation will be done too!

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	domainAPI := domain.NewAPI(p.client)
+	_, err = domainAPI.UpdateDNSZoneRecords(&domain.UpdateDNSZoneRecordsRequest{
+		DNSZone: zone,
+		Changes: []*domain.RecordChange{
+			{
+				Set: &domain.RecordChangeSet{
+					ID: &record.ID,
+					Records: []*domain.Record{
+						{
+							Name: strings.Trim(strings.ReplaceAll(record.Name, zone, ""), "."),
```
Comment (mholt) [inline client.go:140]: Use `libdns` helper functions instead, AbsoluteName() and RelativeName(): https://pkg.go.dev/github.com/libdns/libdns#AbsoluteName

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2020 Matthew Holt
+Copyright (c) <TODO: YEAR AND NAME>
```
[the remark this answers is not in the document]
Comment (dirien) [reply LICENSE:3]: Ok, will fix this in the real repo! Thanks for pointing out!

File: _example/main.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	scaleway "github.com/libdns/digitalocean"
```
[the remark this answers is not in the document]
Comment (dirien) [reply _example/main.go:9]: will fix this. 😅

File: client.go (hunk 1 above)
Comment (dirien) [reply client.go:140]: Yes, will fix this in the real repo!

File: client.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	zoneRecords, err := domainAPI.ListDNSZoneRecords(&domain.ListDNSZoneRecordsRequest{
+		DNSZone: zone,
+	})
+	if err != nil {
+		return records, err
+	}
+
+	for _, entry := range zoneRecords.Records {
+		record := libdns.Record{
+			Name:  entry.Name + "." + strings.Trim(zone, ".") + ".",
```
[the remark this answers is not in the document]
Comment (dirien) [reply client.go:53]: hi @mholt,

done! Hope I did everything right.

**Pull request closed without merging**